LaTeX/LaTeX.sublime-syntax

```diff
@@ -110,67 +118,120 @@ contexts:
 
   macros:
     - include: scope:text.tex#macros
-    - match: |-
-        (?x)
-        (
-          (\\)
-          (?:(?:new|renew|provide)command\*?)
-        )
-        (?:
-          (\{)(\\[A-Za-z@]+)(\})
-          | (\\[A-Za-z@])+
-        )
-        (?:(\[)(?:[^\]]*)(\]))*
-        (\{)
+    - match: '(\\)(?:new|renew|provide)command\*?'
+      scope: meta.function.latex keyword.declaration.function.latex storage.modifier.newcommand.latex
       captures:
-        1: support.function.newcommand.latex storage.modifier.newcommand.latex
-        2: punctuation.definition.backslash.latex
-        3: punctuation.definition.group.brace.begin.latex
-        4: support.function.latex entity.name.newcommand.latex
-        5: punctuation.definition.group.brace.end.latex
-        6: support.function.latex
-        7: punctuation.definition.group.bracket.begin.latex
-        8: punctuation.definition.group.bracket.end.latex
-        9: punctuation.definition.group.brace.begin.latex
+        1: punctuation.definition.backslash.latex
       push:
-        - meta_scope: meta.function.newcommand.latex
-        - match: '\}'
-          scope: punctuation.definition.group.brace.end.latex
-          pop: true
+        - newcommand-definition
+        - newcommand-optarg
+        - newcommand-argspec
+        - newcommand-commandname
+    - match: '(\\)DeclareMathOperator\*?'
+      scope: meta.function.latex support.function.declare-math-operator.latex storage.modifier.newcommand.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      push:
+        - newcommand-definition
+        - newcommand-commandname
+
+  newcommand-argspec:
+    - include: paragraph-pop
+    # direct specification of the argument count: This is what we usually expect,
+    # and what we want to support well
+    - match: '(\[)\s*(\d)\s*(\])'
+      scope: meta.function.parameters.latex
+      captures:
+        1: punctuation.definition.group.bracket.begin.latex
+        2: constant.numeric.value.latex meta.number.integer.decimal.latex
+        3: punctuation.definition.group.bracket.end.latex
+      pop: true
+
+    # however, we might also get weirder argument specifications. Most of them
+    # will be wrong, but determining that would be very difficult, so we're just
+    # very lenient here and accept anything enclosed in brackets.
+    - match: '\['
+      scope: punctuation.definition.group.bracket.begin.latex
+      push:
+        - meta_scope: meta.function.parameters.latex
+        - match: '\]'
+          scope: punctuation.definition.group.bracket.end.latex
+          pop: 2
```

Comment:
Adding a pattern to match a number within brackets could help to avoid duplicated patterns for brackets (by being able to remove the above one L#142-148).

```suggestion
        - match: '\]'
          scope: punctuation.definition.group.bracket.end.latex
          pop: 2
        - match: \d+\b
          scope: meta.number.integer.decimal.latex constant.numeric.value.latex
```

Just considder a construct like ...

```latex
\newcommand{\foo}[
  1
]
```

Reply:
I've applied this change, and also updated the test to have the [ and number on different lines to check this case.